### PR TITLE
Add support for `flags2`

### LINKED ISSUE #365: Add `flags2` support
Due to the bits of `flags` have been exhausted, `flags2` has now appeared in Layer 140.

```typelanguage
channelFull#ea68a619 flags:# can_view_participants:flags.3?true can_set_username:flags.6?true can_set_stickers:flags.7?true hidden_prehistory:flags.10?true can_set_location:flags.16?true has_scheduled:flags.19?true can_view_stats:flags.20?true blocked:flags.22?true flags2:# can_delete_channel:flags2.0?true id:long about:string participants_count:flags.0?int admins_count:flags.1?int kicked_count:flags.2?int banned_count:flags.2?int online_count:flags.13?int read_inbox_max_id:int read_outbox_max_id:int unread_count:int chat_photo:Photo notify_settings:PeerNotifySettings exported_invite:flags.23?ExportedChatInvite bot_info:Vector<BotInfo> migrated_from_chat_id:flags.4?long migrated_from_max_id:flags.4?int pinned_msg_id:flags.5?int stickerset:flags.8?StickerSet available_min_id:flags.9?int folder_id:flags.11?int linked_chat_id:flags.14?long location:flags.15?ChannelLocation slowmode_seconds:flags.17?int slowmode_next_send_date:flags.18?int stats_dc:flags.12?int pts:int call:flags.21?InputGroupCall ttl_period:flags.24?int pending_suggestions:flags.25?Vector<string> groupcall_default_join_as:flags.26?Peer theme_emoticon:flags.27?string requests_pending:flags.28?int recent_requesters:flags.28?Vector<long> default_send_as:flags.29?Peer available_reactions:flags.30?Vector<string> = ChatFull;
```

https://github.com/telegramdesktop/tdesktop/blob/3e5b52bc2e498f45b13729b29197e07be1da9623/Telegram/Resources/tl/api.tl#L132

The regex to parse `flags2` is fixed in #308. However, arguments related to `flags2` are not actually parsed correctly, and `flags2` are not written correctly.

P.S. I think instead of adding support for `flags2`, it might be better allow `flags` to be named anything in the future.

### PR DESCRIPTION
**Pull request by Coxxs**
Fix #365

### CODE REVIEW

**Comment by painor** [comment]
this is a good step but it might need to be tested a bit more before I'm comfortable with merging it.

Have you tested it with ChannelFull yet ?

**Comment by Coxxs** [comment]
Here's the test:

```
import { Api } from 'telegram'
import { BinaryReader } from "telegram/extensions/BinaryReader.js";
import bigInt from "big-integer";

let obj1 = new Api.ChannelFull({
    canViewParticipants: false,
    canSetUsername: false,
    canSetStickers: false,
    hiddenPrehistory: false,
    canSetLocation: false,
    hasScheduled: false,
    canViewStats: false,
    blocked: false,
    canDeleteChannel: true, // flags2 = 1
    id: bigInt('1006503122'),
    about: "Thoughts from the Product Manager / CEO / Founder of Telegram.",
    participantsCount: 657355,
    readInboxMaxId: 0,
    readOutboxMaxId: 0,
    unreadCount: 0,
    chatPhoto: new Api.Photo({
      hasStickers: false,
      id: bigInt('155759888548607294'),
      accessHash: bigInt('-12334823552140358'),
      fileReference: Buffer.from('00eb41e86d6e537ea20f6c4f102faaa353382a3a0a', 'hex'),
      date: 1446056956,
      sizes: [
        new Api.PhotoSize({
          type: "a",
          w: 160,
          h: 160,
          size: 7099
        }),
        new Api.PhotoSize({
          type: "b",
          w: 320,
          h: 320,
          size: 18943
        }),
        new Api.PhotoSize({
          type: "c",
          w: 640,
          h: 640,
          size: 49048
        })
      ],
      dcId: 1
    }),
    notifySettings: new Api.PeerNotifySettings({}),
    botInfo: [],
    pts: 542,
    availableReactions: ["👍", "👎", "🎉"]
});

const buf1 = obj1.getBytes()
console.log(buf1)

const obj2 = (new BinaryReader(buf1)).tgReadObject()
const buf2 = obj2.getBytes()
console.log('compare', Buffer.compare(buf1, buf2) === 0)

/*
Before the fix:
<Buffer 
19 a6 68 ea // channelFull#ea68a619
05 00 00 40 // ❌ flags:# kicked_count:flags.2?int banned_count:flags.2?int is undefined, 0x4 should not be set
05 00 00 40 // ❌ flags:# printed again
d2 04 fe 3b 00 00 00 00 // id:long
... >
TypeNotFoundError: Could not find a matching Constructor ID for the TLObject that was supposed to be
        read with ID 456242494. Most likely, a TLObject was trying to be read when
         it should not be read. Remaining bytes: 168
*/

/*
After the fix:
<Buffer 
19 a6 68 ea // channelFull#ea68a619
01 00 00 40 // flags:#
01 00 00 00 // flags2:#
d2 04 fe 3b 00 00 00 00 // id:long
... >
compare true
*/
```

**Comment by painor** [comment]
ok I have finally finished testing the PR.
everything seems to be working fine so good work.

1) instead of using `startsWith("flags")` you can instead use `flagIndicator` which will also return true.

2) can you generate the new api.d.ts file using the new template. To do so just CD into the `\gramjs\gramjs\tl\types-generator` folder and then do `ts-node generate.js`

don't forget to run `prettier --write .` at the root folder after you finish.

if you want me to do these cleanups instead I'll be happy to.

File: gramjs/tl/api.d.ts [hunk 1]
```diff
@@ -5088,15 +5088,15 @@ export namespace Api {
     }
     export class ReceivedNotifyMessage extends VirtualClass<{
         id: int;
-        // flags: int;
+        flags: int;
```
Comment (Coxxs) [inline gramjs/tl/api.d.ts:5091]: The type of flags here is `flags:int`, which is not a flag indicator (`flags:#`).
By changing `=== 'flags'` to `.flagIndicator`
 in the last commit accidently fixed this :)

**Comment by Coxxs** [comment]
Thanks for testing, I've fixed the above issues.

File: gramjs/tl/api.d.ts (hunk 1 above)
Comment (Coxxs) [reply gramjs/tl/api.d.ts:5091]: But the `flags` here is "Reserved for future use", which is useless now... If user doesn't provided the `flags`, will this cause an Error?

Comment (painor) [reply gramjs/tl/api.d.ts:5091]: this is a result from telegram so the user will not be sending it anyway. we can ignore it for now.

**Pull request merged**